### ISSUE TITLE
OnApply/OnSelected callback

### ISSUE THREAD

**Issue by AirHater** [MemeMayhem/ModExamples#111]
Requesting On Perk/Relic Selected  and/or  On Perk/Relic Applied callback

**Comment by GoodCrispyShark** [comment]
so you want a callback for perk A that can trigger when player gets perk B/C/D?
or do you simply want a callback for on_apply for perk A?

**Comment by AirHater** [comment]
originally i wanted to be able to replicate Wish behaviour, but since its hardcoded and CombatUnit:CombineRelics doesnt have combo_data.required_perks as an input (i am also not even sure if RegisterOnExitPerkSelectCallback would even register everything properly), i wanted to have some proper callback for these types of perks (or relics), that have effect on acquiring other perks (or relics), animation of combining included

**Comment by GoodCrispyShark** [comment]
i will try to add support for this either this week or next week

**Comment by AirHater** [comment]
is it possible to, at least temporary, add required perks to CheckRelicCombos() / CheckCanCombine() / CombineRelics() in CombatUnit, with something like `formula = {
relics = {relic_1 = 1,},
perks = {perk_1 = 1},
perks_removed = {perk_1 = 1}
}` for their formula?

**Comment by GoodCrispyShark** [comment]
im not sure if i understand your words here
do you mean you want a dynamic combo relic receipt pool that could be changed at run time?

**Comment by AirHater** [comment]
no, i dont need it to be dynamic during runtime, i suggested to make combo relics be able to made from perks (by adding required_perks to CombatUnit:CombineRelics() and its dependencies), since layout:Animate() handles them already.
and be able to list whether these perks are removed or not in their formula

**Comment by GoodCrispyShark** [comment]
i see, im not 100% sure if that could work but i could take a look when i get time

**Comment by AirHater** [comment]
any update on this?

**Comment by GoodCrispyShark** [comment]
sorry we are working on game updates recently so didnt get time to work on this

**Comment by GoodCrispyShark** [comment]
So you are asking being able to combo relic with perk, and add a variable to adjust whether the perk persists afterwards?

**Comment by AirHater** [comment]
@GoodCrispyShark the reason why i created this issue was indeed to be able to make combo relics with perks (just perks, but having perks + relics for the reqs would be good), and to also be able to remove them (or keep them, as Wish is currently created).
on top of that, it would be good to have callback from acquiring perks, to imitate Fish relics behaviour

**Comment by Meme631** [comment]
@AirHater Hi, we are going to add perk/relic on_apply callbacks. If you want to make relic combo from perks, you can do this as a workaround:
1. Create a perk that will give a dummy relic when this perk is applied(using the new API)
2. Use this dummy relic fro the combo. If you don't want to remove the relic for combo, you can give this dummy relic back when the combo relic is applied(using the new API)

**Comment by AirHater** [comment]
@Meme631 thanks for adding that callback, but i dont think this solves my issue. the animation of combining would be just these dummy relics, not the perks "that are combining", and it would need to have a dupe relic for every "combinable" perk
